tools/Dashboard/DevHome.Dashboard/Helpers/WidgetHelpers.cs

```diff
@@ -73,12 +74,22 @@ public static bool IsIncludedWidgetProvider(WidgetProviderDefinition provider)
         return include;
     }
 
-    public static string CreateWidgetCustomState()
+    public static string CreateWidgetCustomState(int ordinal)
     {
         var state = new WidgetCustomState
         {
             Host = DevHomeHostName,
+            Position = ordinal,
         };
         return JsonSerializer.Serialize(state);
     }
+
+    public static async Task SetPositionOnWidget(Widget widget, int ordinal)
+    {
+        var stateStr = await widget.GetCustomStateAsync();
+        var stateObj = JsonSerializer.Deserialize<WidgetCustomState>(stateStr);
+        stateObj.Position = ordinal;
+        stateStr = JsonSerializer.Serialize<WidgetCustomState>(stateObj);

```

Comment:
These two serialization calls are not trim/AOT-safe. Consider rebasing on top of #https://github.com/microsoft/devhome/pull/1204 and pass the `JsonSerializerContext` object here instead 🙂